test_support/src/lib.rs

```diff
@@ -119,6 +120,27 @@ pub fn assert_web_response(ctx: &TestContext, expected_response_body: &'static s
     });
 }
 
+pub fn wait_for<F>(condition: F, max_wait_time: Duration)
+where
+    F: Fn() + panic::RefUnwindSafe,
+{
+    let start_time = SystemTime::now();
+    while SystemTime::now()
+        .duration_since(start_time)
+        .expect("should not be an earlier time")
+        < max_wait_time
+    {
+        let result = panic::catch_unwind(|| {
+            condition();
+        });
+        if result.is_ok() {
+            return;
+        }
+        std::thread::sleep(Duration::from_millis(10));
+    }
+    panic!("timeout exceeded");
```

Comment:
Would we ever see the actual failure message while using `wait_for`? What I mean is, if this check continually fails:

```
assert_not_contains!(
  container.logs_now().stderr,
  "Registering metrics instrumentation"
);
```
Not because of a timeout, but because we changed the message, does that error show up? Or is only the `timeout exceeded` message seen?

Reply:
Just thinking it might make more sense to do something like this after the timeout:

```
eprintln!("wait_for timeout exceeded");
condition()
```

So we still get the original assertion messages, and can see that it timed out.

Reply:
Good call.  I'll capture the potential error from the condition and propagate it forward so if the desired outcome wasn't reached within the timeout it will print the failure message from `assert_not_contains!(container.logs_now().stderr, "Registering metrics instrumentation");`